lib/ansible/modules/cloud/docker/docker_swarm.py

```diff
@@ -230,6 +230,12 @@
                   returned: success
                   type: str
                   example: SWMTKN-1--xxxxx
+      UnlockKey:
+          description: The swarm unlock-key if I(autolock_managers) is C(true).
+          returned: success
```

Comment:
Shouldn't this be something like `returned: on success if I(autolock_managers) is C(true)`? Also, I thought you only want to return it if the swarm was actually created? Maybe it is better to relax that condition, and always return it when the module is called with `state == present` and `autolock_managers == true`. Then the rest of the playbook doesn't has to know whether the swarm was actually created or not.

Reply:
It does not make sense to only return it when the swarm is created. If `autolock_managers` is set to `true` by an update one would expect the key to be returned. This is also the way the docker cli works. In other words if the swarm is created without a lock and later updated with autolock the key is returned. If the update is repeated with autolock the key is not returned.

Returning the key only on changes will reduce the runtime of the task and make it a tiny bit safer as the key is not always returned and logged.

Reply:
It does make sense to return when the swarm is created or when the key is changed. In any other case, it can be read by `docker_swarm_info`. Actually, I think the plan was to remove all `inspect` features from `docker_swarm`. @felixfontein do we have time to do that?

Reply:
I'm mainly concerned that the condition when that value is returned might be too confusing for users who want to use it. But in the end I'm fine when it's properly documented when exactly it is returned, and when a pointer is added to `docker_swarm_info` so that people know they can get it reliably from there.

@WojciechowskiPiotr we can deprecate `inspect` (in fact we already did that), but it has to stay until Ansible 2.12.

Reply:
I added documentation about that `docker_swarm_info` can be used to retrieve the unlock key under the `autolock_managers` option.